hack/test-cmd.sh

```diff
@@ -646,7 +646,7 @@ runTests() {
   # Command: autoscale rc "frontend" 
   kubectl autoscale -f examples/guestbook/frontend-controller.yaml --save-config "${kube_flags[@]}" --max=2
   # Post-Condition: hpa "frontend" has configuration annotation
-  [[ "$(kubectl get hpa frontend -o yaml "${kube_flags[@]}" | grep kubectl.kubernetes.io/last-applied-configuration)" ]]
+  [[ "$(kubectl get hpa.extensions frontend -o yaml "${kube_flags[@]}" | grep kubectl.kubernetes.io/last-applied-configuration)" ]]
```

Comment:
> I'd rather see a block of calls that exercise all the possibilities, instead of scattering them through the file. As we add things that make kinds ambiguous we could update that section to test getting the kinds in the new groups, and make sure the unqualified kinds still returned the old group

The actual working bits of code are fully exercised in unit tests and I don't know of a conflicting resource that we have right now.  Once we have one, such a test could be written, but I don't have many options now.  Duplicating tests doesn't seem like a great idea.